### PR TITLE
chore: update web3-utils

### LINKED ISSUE #8747: Dependencies: web3-utils
- update `web3-utils` to `4.0.1`+

**Issues:**
- failing ETH and XRP tests in `useSendForm.test.tsx`
- fails because `TextEncoder`, `Uint8Array` is missing and a problem with `waitForElementToBeRemoved` method
- 9 tests are failing, after updating jest to `29` #8726, the number reduces to 6 
Screenshot 2023-06-20 at 15 14 18
Screenshot 2023-06-20 at 15 14 37

- TextEncoder is available in web environment, but not node. So the app is working okay, but tests are not
https://github.com/web3/web3.js/blob/023f02d890a623ba800c3033074d553fe5346971/packages/web3-utils/src/converters.ts#L231

### PR DESCRIPTION
**Pull request by mroz22**
Resolves #8747

we might need some help from mobile team. this update breaks android e2e. cc @Nodonisko might know 🙏 

```
Error: While trying to resolve module `web3-validator` from file `/home/runner/work/trezor-suite/trezor-suite/node_modules/web3-utils/lib/commonjs/validation.js`, the package `/home/runner/work/trezor-suite/trezor-suite/node_modules/web3-validator/package.json` was successfully found. However, this package itself specifies a `main` module field that could not be resolved (`/home/runner/work/trezor-suite/trezor-suite/node_modules/web3-validator/dist/web3-validator.min.js`. Indeed, none of these files exist:
```

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -40162,7 +40171,14 @@ __metadata:
...
 
-"web3-utils@npm:1.10.3, web3-utils@npm:^1.10.0":
+"web3-types@npm:^1.3.1, web3-types@npm:^1.5.0, web3-types@npm:^1.6.0":
+  version: 1.6.0
+  resolution: "web3-types@npm:1.6.0"
+  checksum: 10/9eac2082080c7f73777f57d688041ed4938b0943ff0ece7113c220bd85ddd7ce1191969302bbfbe31464907639270fc54d8e5cd7a7f200262ede81516dda39f9
+  languageName: node
+  linkType: hard
+
+"web3-utils@npm:1.10.3":
```
Comment (legobeat) [inline yarn.lock:40181]: To fully get rid of web3 v1, an upstream update in [`@everstake/wallet-sdk`](https://github.com/everstake/wallet-sdk) (used in [`@trezor/suite`](https://github.com/trezor/trezor-suite/blob/990fc2ceeba16bf3c24ca621f88703f693c9afc4/packages/suite/package.json#L22)) is needed.

https://github.com/everstake/wallet-sdk/issues/11

**Comment by Nodonisko** [comment]
Nice mobile e2e finally pays off :D

**Review by tomasklim — APPROVED** [review]
LGTM tested

**Comment by Nodonisko** [comment]
@PeKne will take care of it.

**Comment by PeKne** [comment]
Native build (E2E) fixed by [088dc2e](https://github.com/trezor/trezor-suite/pull/12184/commits/088dc2eeaed3612e1c6bbcf6a22324210ba3f62c).

**Comment by mroz22** [comment]
@SocketSecurity ignore-all

**Pull request merged**